### PR TITLE
Added iframephone calls to inform LARA if document is dirty [#130837611]

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

**Comment by dougmartin** [comment]
@scytacki For review with https://github.com/concord-consortium/lara/pull/238

File: app/views/snapshots/_form.html.haml [hunk 1]
```diff
@@ -72,6 +72,62 @@
...
+              draggable: false,
+              modal: true,
+              resizable: false,
+              width: 'auto',
+              title: 'Confirm',
+              minHeight: 75,
+              buttons: {
+                Yes: function () {
+                  canClose = true;
+                  phone.post('disconnect');
```
Comment (scytacki) [inline app/views/snapshots/_form.html.haml]: I don't see a handler for disconnect in LARA.  Is it necessary to disconnect in this case?
If connected LARA will send the canClose message. And then the labbook will immediately respond with 'close'.

**Review by scytacki — CHANGES_REQUESTED** [review]
seems like disconnect message isn't handled.

**Comment by scytacki** [comment]
This looks good to me. @knowuh do you want to take a quick look?

**Comment by knowuh** [comment]
Took me a while to figure out what this was all about. In the end, it looks fine to me.  👍

**Review by scytacki — APPROVED** [review]

**Comment by scytacki** [comment]
OK I'm going to wait to merge this until a small UI fix is in regarding a browse button on firefox.

**Comment by scytacki** [comment]
Heroku staging server is having trouble building this. I haven't investigated yet.

**Pull request merged**